gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -433,18 +483,27 @@ public static List<Service> parseService(
       Optional<GapicServiceConfig> serviceConfigOpt,
       Set<ResourceName> outputArgResourceNames,
       Transport transport) {
-
+    String protoPackage = fileDescriptor.getPackage();
+    Optional<List<String>> inclusionMethodListFromServiceYaml =
+        getInclusionMethodListFromServiceYaml(serviceYamlProtoOpt, protoPackage);
     return fileDescriptor.getServices().stream()
         .filter(
             serviceDescriptor -> {
               List<MethodDescriptor> methodsList = serviceDescriptor.getMethods();
-              if (methodsList.isEmpty()) {
+              List<MethodDescriptor> methodListSelected =
```

Comment:
This scenario is in case that all methods from a service are excluded?

Reply:
Yes. And this is likely now e.g. in vertexai's requirement it only exposes methods from 2 services.